junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeout.java

```diff
@@ -141,8 +148,11 @@ static void assertTimeoutPreemptively(Duration timeout, Executable executable, S
 				return future.get(timeoutInMillis, TimeUnit.MILLISECONDS);
 			}
 			catch (TimeoutException ex) {
+				ExecutionTimeoutException exception = new ExecutionTimeoutException("Execution timed out");
+				exception.setStackTrace(threadReference.get().getStackTrace());
```

Comment:
I think we should handle the case of `threadReference` not being set, just to be safe.

Reply:
Should I test the `null` `threadReference` case? If so how? 🤔

Reply:
The only thing we could do is to add a package-private overload that takes a `ExecutorService` but I don't really think it's worth it in this corner case.